### DISCUSSION TITLE
Change in energy if restarted with the last configuration.

### DISCUSSION THREAD

**Discussion by mubisk** [glotzerlab/hoomd-blue#1693] [Support] [answered]
Hi,  I gave a run and save the configurations along with the thermodynamic properties in a gsd file. Then, again the runs were given initializing with the last configuration. The energy profile of the system was well equiliberated in the previous run but in the next run, the value of energy starts with a bit higher energy and then increases and takes a few hundred runs to get back to the same value again. Though it quickly comes back to the same value but this is not only unexpected but also looks erroneous. Does any body has a similar experience or can point out the issue in this problem??

**Comment by joaander** [comment]
Please provide a minimal script that demonstrates this behavior.

**Comment by mubisk** [comment]
Hi. Thanks for the reply. Please find the below code:

for ppp in range(0,1):

    #temp = temp - 0.03
    cpu = hoomd.device.GPU()
    sim = hoomd.Simulation(device=cpu, seed=7376)
    sim.create_state_from_gsd(filename='../Sim_k=0.0.gsd',frame=-1)
    L_bond=1.0
    k_v = 0.0
    fene = md.bond.FENEWCA()
    fene.params['A-A'] = dict(k=30.0, r0=1.5, sigma=1.0, epsilon=1.0, delta=0.0)
    harmonic_angle = md.angle.Harmonic()
    harmonic_angle.params['A-A-A'] = dict(k=k_v, t0=3.141592)
    nl = md.nlist.Cell(buffer=1.0)
    fslj = md.pair.ExpandedLJ(default_r_cut=1.12246*L_bond, nlist=nl, mode='shift')
    fslj.params[('A', 'A')] = dict(epsilon=1.0, sigma=L_bond, delta=0.0)
    integrator = md.Integrator(dt=0.005)
    langevin = md.methods.Langevin(filter=hoomd.filter.All(), kT=temp)
    integrator.methods.append(langevin)
    integrator.forces.append(fslj)
    integrator.forces.append(fene)
    integrator.forces.append(harmonic_angle)
    thermodynamic_properties = hoomd.md.compute.ThermodynamicQuantities(filter=hoomd.filter.All())
    sim.operations.computes.append(thermodynamic_properties)
    logger = hoomd.logging.Logger()
    logger.add(thermodynamic_properties)
    out_gsd = hoomd.write.GSD(filename="Sim_k="+str(round(k_v,1))+".gsd", trigger=100, mode='ab', truncate=False, dynamic=['property'], log=logger, filter=hoomd.filter.All())
    sim.operations.writers.append(out_gsd)
    out_gsd.log = logger
    sim.operations.integrator = integrator
    sim.run(50000)

**Comment by tcmoore3** [comment, marked as answer]
Without more information (e.g., is it the kinetic or potential energy that changes upon restart?), it is impossible to be sure why you observe this behavior. However, the issue is most likely that you are not saving particle velocities (see https://hoomd-blue.readthedocs.io/en/latest/module-hoomd-write.html#hoomd.write.GSD), so when you create the simulation state all of the velocities are set to the default value of 0. Since you do not thermalize the system momwntum (https://hoomd-blue.readthedocs.io/en/latest/package-hoomd.html#hoomd.State.thermalize_particle_momenta), the system is starting at temperature = 0, and it takes the thermostat a few steps to get the system to the set temperature.

**Reply by joaander** [reply]
Yes, this is the cause of the behavior. The `hoomd.write.GSD(dynamic=['property'], ..)` does not save velocity on every frame. The [tutorial on continuing simulations](https://hoomd-blue.readthedocs.io/en/v4.4.1/tutorial/05-Organizing-and-Executing-Simulations/03-Continuing-Simulations.html) suggests using separate restart files written by `hoomd.write.GSD.write` to store all necessary state information.

**Comment by mubisk** [comment]
Yes, Exactly thats what was the problem. Thanks a lot.
